tests/test_digital.py

```diff
@@ -322,3 +323,17 @@ def test_dm(n_qubits: Tensor, batch_size: Tensor) -> None:
     dm = density_mat(state_cat)
     assert dm.size() == torch.Size([2**n_qubits, 2**n_qubits, batch_size])
     assert torch.allclose(dm, dm_proj)
+
+
+size = (5, 2)
+random_param = torch.randperm(size[0] * size[1])
+random_param = random_param.view(size)
+random_param = torch.sort(random_param, dim=1)[0]
+
+
+@pytest.mark.parametrize("target,n_qubits", random_param)
+def test_promote(target: int, n_qubits: int) -> None:
+    I_prom = promote_ope(I(0).unitary(), target, n_qubits)
+    assert I_prom.size() == torch.Size([2**n_qubits, 2**n_qubits, 1])
+    X_prom = promote_ope(X(0).unitary(), target, n_qubits)
+    assert X_prom.size() == torch.Size([2**n_qubits, 2**n_qubits, 1])
```

Comment:
In this test, I test only the size of the output tensor. Even if the goal is to scale up the tensor, I think testing the tensor values is also important. 
However, I cannot find a way to do this test without almost re-coding the function itself as shown in the following example. What do you think? 

``` python3
size = (5, 2)
random_param = torch.randperm(size[0] * size[1])
random_param = random_param.view(size)
random_param = torch.sort(random_param, dim=1)[0]


@pytest.mark.parametrize("target,n_qubits", random_param)
def test_promote(target: int, n_qubits: int) -> None:
    #Size test:
    identity = I(target).unitary()
    I_prom = promote_ope(identity, target, n_qubits)
    assert I_prom.size() == torch.Size([2**n_qubits, 2**n_qubits, 1])
    flip = X(target).unitary()
    X_prom = promote_ope(flip, target, n_qubits)
    assert X_prom.size() == torch.Size([2**n_qubits, 2**n_qubits, 1])

    #Value test
    qubits = torch.arange(0, n_qubits)
    for qubit in qubits:
        if target > qubit:
            flip = torch.kron(I(qubit).unitary(), flip)
            identity = torch.kron(I(qubit).unitary(), identity)
        elif target < qubit:
            flip = torch.kron(flip, I(qubit).unitary())
            identity = torch.kron(identity, I(qubit).unitary())
    assert torch.allclose(promote_ope(X(target).unitary(),target,n_qubits),flip)
    assert torch.allclose(promote_ope(I(target).unitary(),target,n_qubits),identity)
    assert torch.allclose(promote_ope(I(target).unitary(),target,n_qubits),torch.eye(2**n_qubits,dtype=torch.cdouble).unsqueeze(2))
```

Reply:
Could be. One way could also to define a parametrization as a list of gates and to check them all. Maybe one simple way would be to check that `gate * gate.adjoint() == I` ?

Reply:
What do you think of this version ? 
```python3
from pyqtorch.primitive import I,X,Y,Z,H,T,S
from pyqtorch.matrices import _dagger

size = (5, 2)
random_param = torch.randperm(size[0] * size[1])
random_param = random_param.view(size)
random_param = torch.sort(random_param, dim=1)[0]

@pytest.mark.parametrize("target,n_qubits", random_param)
@pytest.mark.parametrize("operator", [I,X,Y,Z,H,T,S])
def test_promote(target: int, n_qubits: int, operator: Tensor) -> None:
    op_prom = promote_op(operator(target).unitary(), target, n_qubits)
    assert op_prom.size() == torch.Size([2**n_qubits, 2**n_qubits, 1])
    assert torch.allclose(
        apply_ope_ope(op_prom,_dagger(op_prom),target),torch.eye(2**n_qubits,dtype=torch.cdouble).unsqueeze(2)
        )